doc/api/esm.md

```diff
@@ -586,12 +558,27 @@ an all-ES module-syntax version the package. This could be used via `import
 'pkg/module'` by users who are certain that the CommonJS version will not be
```

Comment:
We could change the wording in this paragraph to `A variant of this approach not requiring `--experimental-conditional-exports` for consumers can be to ...`

Perhaps we should also note that the variation can progressively enhance to use conditional exports unflagged in future too.